src/domain/discounts/index.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useRef, useState, useContext } from "react"
+import qs from "query-string"
+import { Router } from "@reach/router"
+import New from "./new"
+import Details from "./details"
+
+import useMedusa from "../../hooks/use-medusa"
+
+import Spinner from "../../components/spinner"
+import { navigate } from "gatsby"
+import { InterfaceContext } from "../../context/interface"
+import PageDescription from "../../components/atoms/page-description"
+import BodyCard from "../../components/organisms/body-card"
+import PlusIcon from "../../components/fundamentals/icons/plus-icon"
+import DiscountTable from "../../components/templates/discount-table"
+
+const DiscountIndex = () => {
+  const filtersOnLoad = qs.parse(window.location.search)
+
+  if (!filtersOnLoad.offset) {
+    filtersOnLoad.offset = 0
+  }
+
+  if (!filtersOnLoad.limit) {
+    filtersOnLoad.limit = 20
+  }
+
+  const { discounts, refresh, isReloading, isLoading } = useMedusa(
+    "discounts",
+    {
+      search: {
+        is_disabled: "false",
+        is_dynamic: "false",
+        ...filtersOnLoad,
+      },
+    }
+  )
```

Comment:
Should use `useAdminDiscounts` hook

Reply:
Do we need this? As far as I see, we are doing the fetching in the actual discount table component. If not, we should remove this and all of the below querying and filtering :)